src/WebJobs.Script/Diagnostics/FileLogger.cs

```diff
@@ -0,0 +1,98 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using Microsoft.Azure.WebJobs.Host;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Script.Diagnostics
+{
+    /// <summary>
+    /// A wrapper class to allow ILoggers from functions to write to the existing FileTraceWriter. This allows
+    /// logs to show up in the log files and stream in the portal.
+    /// </summary>
+    internal class FileLogger : ILogger
```

Comment:
Well, my original PR said "tests coming soon" -- I forgot to put it into this PR. Yes, they're coming.